homework/list.js

```diff
@@ -52,28 +60,34 @@
       buff.appendChild(li);
     });
     ul.appendChild(buff);
-    _wrapper.appendChild(ul);
+    this._wrapper.appendChild(ul);
+    
   }
 
-  function fetchList(afterFetch) {
+  listObject.prototype.fetchList = function() {
+  	return new Promise(function(resolve, reject){ 
     var xhr = new XMLHttpRequest();
-    xhr.open('GET', 'http://127.0.0.1:8000/demo-list-notes.json', true);
+    xhr.open('GET', 'http://127.0.0.1:8000/homework/demo-list-notes.json', true);
     xhr.responseType = 'json';
     xhr.onreadystatechange = function(e) {
       // Watch out: we have a mysterious unknown 'this'.
       if (this.readyState === 4 && this.status === 200) {
         var listData = this.response;
         // The flow ends here.
-        afterFetch(listData);
+        //afterFetch(listData);
+        resolve(listData);
       } else if (this.status !== 200 ){
         // Ignore error in this case.
+        reject();
       }
     };
     xhr.send();
+  })
   }
 
   document.addEventListener('DOMContentLoaded', function(event) {
-    start();
+	 var begin = new listObject();    
```

Comment:
**Please be more serious about your coding style.** Naming things like this is just to mess up your project and make long term maintenance impossible.